public/app/core/context/ModalsContextProvider.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+
+import { textUtil } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { ConfirmModal, ConfirmModalProps, ModalsContext } from '@grafana/ui';
+import { ModalsContextState } from '@grafana/ui/src/components/Modal/ModalsContext';
+import { ShowConfirmModalEvent, ShowModalReactEvent } from 'app/types/events';
+
+import appEvents from '../app_events';
+
+export interface Props {
+  children: React.ReactNode;
+}
+
+/**
+ * Implements the ModalsContext state logic (not used that much, only needed in edge cases)
+ * Also implements the handling of the events ShowModalReactEvent and ShowConfirmModalEvent.
+ */
+export function ModalsContextProvider(props: Props) {
+  const [state, setState] = useState<ModalsContextState>({
+    component: null,
+    props: {},
+    showModal: (component: React.ComponentType<any>, props: any) => {
+      setState({ ...state, component, props });
+    },
+    hideModal: () => {
+      setState({ ...state, component: null, props: {} });
+    },
+  });
+
+  useEffect(() => {
```

Comment:
Lack of depedency array to this effect means it'll re-run, and re-subscribe each time it's rerendered. I'm surprised this isn't already an issue, but I think this could be the source of subtle bugs in the future.

I think we should just pass in the empty dep array.

Reply:
fixed